data_filter_azure/README.containerregistry.md

```diff
@@ -0,0 +1,108 @@
+
+The server itself is implemented in Python using Flask. 
+
+## Running the sample
+
+The is only prototype and doesn't include the authentication. The installation and the model is adopted from the existing sample under data_filter_example of this project
+
+## Data and Policy
+
+A resource called "registry" can contain different types of resources with different supported actions against them. Each authenticated user will have the mappings of allowed resources and actions. For example the below document represents the mappings for a registry owned by user 'bob'. The data represents that Bob can read, write test-repo and read/delete test-chart. 
+
+```json
+{
+    "bob": {
+        "repositories": {
+            "repo1": {
+                "actions": ["read", "write"]
+            },
+            "repo2": {
+                "actions": ["*"]
+            }
+        },
+        "charts": {
+            "chart1": {
+                "actions": ["read", "write"]
+            }
+        },
+        "pipelines": {
+            "*": {
+                "actions": ["read"]
+            }
+        }
+    }
+}
+```
+
+## Running the sample
+
+1. Create the bundle with data and polcies. Create the folder structure with corresponding data and policy rego files.
+
+```bash
+export BUNDLE_TAR=registries.tar.gz
+cd opabundle
+tar -czvf $BUNDLE_TAR .
+```
+
+2. Push the bundle to the Azure container registry. Create an [azure container registry](https://azure.microsoft.com/en-us/services/container-registry/)
+
+3. The sample fetchs the container registry keys and login server from environment variables ACR_LOGINSERVER, ACR_USERNAME and ACR_PASSWORD.
+
+```bash
+export ACR_LOGINSERVER=<login server of azure container registry>
+export ACR_USERNAME=<username of azure container registry>
+export ACR_PASSWORD=<password1 of azure container registry>
+```
+
+4. Install [oras](https://github.com/deislabs/oras) to push the bundle to the container registry and set the environment variable ACR_BUNDLE_ARTIFACT_SHA to the Digest of the bundle. This will be used to pull the bundle from container registry when OPA makes a request for the bundle. 
+
+```bash
+output=$(oras push -u $ACR_USERNAME -p $ACR_PASSWORD $ACR_LOGINSERVER/$BUNDLE_TAR $BUNDLE_TAR)
+export ACR_BUNDLE_ARTIFACT_SHA=$(echo $output | grep -Po 'Digest: \K(.*)')
+```
+
+5. Install the dependencies into a virtualenv:
+
+```bash
+cd ..
+virtualenv env
+source env/bin/activate
+pip install -r requirements.txt
+pip install -e .
+```
+
+6. Start the server:
```

Comment:
How about: "Start the server that downloads bundles from the Microsoft registry and serves those bundles to OPA."